### DISCUSSION TITLE
Object Detection

### DISCUSSION THREAD

**Discussion by camasmie** [introlab/rtabmap#1210] [Q&A] [answered]
I am writing our own motion planning for a robotic arm with collision avoidance. 

I am thinking the most direct way To find the xyz position of obstacles would be to use the octomap output and check if a position is occupied (ignoring positions around the goal object, which I have). 

The Octomap library has: a leaf iterator, a getCoordinate and a isNodeOccupied methods.

Is this a sensible approach, or is there another solution already implemented in rtabmap?

Thanks!

**Comment by matlabbe** [comment, marked as answer]
If your robotic arm is not on a mobile base, you may be able to just use [octomap_server](http://wiki.ros.org/octomap_server) directly. For 3D output format in RTAB-Map, we have raw point cloud, mesh (with standalone library we can generate a mesh) or OctoMap. For your case, I guess the OctoMap makes more sense with its OcTree structure to know space that is occupied, unknown or empty.

**Reply by camasmie** [reply]
Thank you @matlabbe

I will look into it.
